### ISSUE TITLE
Not able to Reproduce results Intrinsic Results Using Data Example

### ISSUE THREAD

**Issue by Kwame261** [google-research/google-research#1566]
Hello,

I am trying to reproduce the model learning intrinsic parameters during training. I used the data example in the repository. I modified the code to print the intrinsic parameters during training as shown below.

  if not params.learn_intrinsics.enabled:
    endpoints['intrinsics_mat'] = features['intrinsics_mat']
    endpoints['intrinsics_mat_inv'] = features['intrinsics_mat_inv']
  elif params.learn_intrinsics.per_video:
    int_mat = intrinsics_utils.create_and_fetch_intrinsics_per_video_index(
        features['video_index'][0],
        params.image_preprocessing.image_height,
        params.image_preprocessing.image_width,
        max_video_index=params.learn_intrinsics.max_number_of_videos)
    intrinsics_mat = tf.Print(intrinsics_mat,[intrinsics_mat],message="This is a: ",summarize=None) 
    

    endpoints['intrinsics_mat'] = tf.concat([int_mat] * 2, axis=0)
    endpoints['intrinsics_mat_inv'] = intrinsics_utils.invert_intrinsics_matrix(int_mat)
    
    
  else:
    # The intrinsic matrix should be the same, no matter the order of
    # images (mat = inv_mat). It's probably a good idea to enforce this
    # by a loss, but for now we just take their average as a prediction for the
    # intrinsic matrix.
    intrinsics_mat = tf.Print(intrinsics_mat,[intrinsics_mat],message="This is a: ",summarize=8)
    intrinsics_mat = 0.5 * sum(tf.split(intrinsics_mat, 2, axis=0))
    
    
    endpoints['intrinsics_mat'] = [intrinsics_mat] * 2
    endpoints['intrinsics_mat_inv'] = [
        intrinsics_utils.invert_intrinsics_matrix(intrinsics_mat)] * 2
From the results below, even though the objective loss seems to be reducing, The focal seems to moving further and further away from the groundtruth[565.63,0.0,222.8240625,0.0,566.3254476497139,64.142125,0.0,0.0,1.0].Please find below the results:

MP: Info #255: KMP_AFFINITY: pid 35976 tid 36120 thread 1 bound to OS proc set 1
This is a: [[[288.060059 0 209.22963][0 88.6649628 63.9474792][0 0...]]...]
INFO:tensorflow:loss = 1.0622927, step = 1
I0412 18:01:38.830058 139998331281600 basic_session_run_hooks.py:262] loss = 1.0622927, step = 1
This is a: [[[285.602539 0 214.193176][0 88.9621506 65.9318466][0 0...]]...]
This is a: [[[285.407623 0 215.046326][0 89.2234192 65.8788147][0 0...]]...]
This is a: [[[285.989655 0 215.964722][0 89.5191269 66.1381378][0 0...]]...]
This is a: [[[285.760437 0 218.972641][0 89.9916763 66.3636093][0 0...]]...]
This is a: [[[285.165741 0 222.539352][0 90.5095367 66.6377792][0 0...]]...]
This is a: [[[284.530701 0 225.656494][0 90.6816177 66.4246292][0 0...]]...]
This is a: [[[284.469177 0 226.012741][0 90.4061508 65.819519][0 0...]]...]
This is a: [[[284.346893 0 225.750824][0 89.984787 65.2077332][0 0...]]...]
This is a: [[[284.610718 0 224.510742][0 89.6685104 64.7842255][0 0...]]...]
This is a: [[[284.77182 0 224.040527][0 89.4978333 64.4307709][0 0...]]...]
This is a: [[[284.782166 0 224.609009][0 89.4429779 64.1097641][0 0...]]...]
This is a: [[[284.871338 0 225.702225][0 89.4668884 63.7531853][0 0...]]...]
This is a: [[[284.377563 0 228.349091][0 89.5583954 63.3299904][0 0...]]...]
This is a: [[[284.404694 0 229.693069][0 89.6804 62.8491058][0 0...]]...]
This is a: [[[284.274628 0 230.50592][0 89.7475662 62.5162926][0 0...]]...]
This is a: [[[284.0224 0 230.659027][0 89.717 62.4574699][0 0...]]...]
This is a: [[[284.322021 0 229.180847][0 89.6059875 62.388298][0 0...]]...]
This is a: [[[284.085052 0 228.497299][0 89.5023651 62.551239][0 0...]]...]
This is a: [[[284.000336 0 227.708359][0 89.3874893 62.6291847][0 0...]]...]
This is a: [[[283.754578 0 227.679][0 89.3105698 62.6709709][0 0...]]...]
This is a: [[[283.680786 0 227.597244][0 89.2733536 62.6397285][0 0...]]...]
This is a: [[[283.512604 0 227.486908][0 89.2438278 62.6564674][0 0...]]...]
This is a: [[[283.305023 0 227.390335][0 89.190567 62.7281227][0 0...]]...]
This is a: [[[283.334869 0 226.616852][0 89.1651917 62.7624626][0 0...]]...]
This is a: [[[283.099304 0 226.881149][0 89.141861 62.8104057][0 0...]]...]
This is a: [[[282.871307 0 227.357178][0 89.1751 62.7699471][0 0...]]...]
This is a: [[[282.929077 0 226.666962][0 89.1495285 62.8052559][0 0...]]...]
This is a: [[[282.688446 0 227.210342][0 89.1257095 62.8216][0 0...]]...]
This is a: [[[282.785065 0 227.23674][0 89.025856 62.8475456][0 0...]]...]
This is a: [[[282.825928 0 226.515762][0 88.9115601 62.9274445][0 0...]]...]
This is a: [[[282.290375 0 227.925766][0 88.8526688 62.9450378][0 0...]]...]
This is a: [[[281.857697 0 229.679596][0 88.8354111 62.8420067][0 0...]]...]
This is a: [[[281.901672 0 229.704102][0 88.8867645 62.7635384][0 0...]]...]
This is a: [[[282.024384 0 229.195023][0 88.8730774 62.6652107][0 0...]]...]
This is a: [[[282.384155 0 227.640488][0 88.8294525 62.6375313][0 0...]]...]
This is a: [[[282.514526 0 226.869965][0 88.8017349 62.6930351][0 0...]]...]
This is a: [[[282.113312 0 228.14711][0 88.781189 62.6561623][0 0...]]...]
This is a: [[[282.090515 0 227.393127][0 88.7616806 62.9868431][0 0...]]...]
This is a: [[[281.779877 0 228.425079][0 88.7520294 62.9765549][0 0...]]...]
This is a: [[[281.329956 0 230.711075][0 88.7534866 62.6900978][0 0...]]...]
This is a: [[[281.562225 0 229.537399][0 88.7111053 62.7359581][0 0...]]...]
This is a: [[[282.012 0 227.587372][0 88.6442108 62.86203][0 0...]]...]
This is a: [[[281.901245 0 227.833099][0 88.5869827 62.8695335][0 0...]]...]
This is a: [[[281.118439 0 229.45433][0 88.5285721 63.0186729][0 0...]]...]
This is a: [[[280.83728 0 230.02681][0 88.4887695 63.2069626][0 0...]]...]
This is a: [[[280.003479 0 232.594315][0 88.4529 63.2158318][0 0...]]...]
INFO:tensorflow:Saving checkpoints for 47 into /tmp/my_experiment22/model.ckpt.
I0412 18:11:11.715969 139998331281600 basic_session_run_hooks.py:606] Saving checkpoints for 47 into /tmp/my_experiment22/model.ckpt.
This is a: [[[279.254028 0 234.986099][0 88.4194717 63.1360054][0 0...]]...]
This is a: [[[279.566193 0 234.513809][0 88.37677 63.1572571][0 0...]]...]
This is a: [[[279.789886 0 231.73056][0 88.3480072 63.4516602][0 0...]]...]
This is a: [[[279.792816 0 233.161011][0 88.2974777 63.1868477][0 0...]]...]
INFO:tensorflow:global_step/sec: 0.0804509
I0412 18:12:00.324321 139998331281600 basic_session_run_hooks.py:692] global_step/sec: 0.0804509
INFO:tensorflow:loss = 0.84261453, step = 51 (621.498 sec)
I0412 18:12:00.328237 139998331281600 basic_session_run_hooks.py:260] loss = 0.84261453, step = 51 (621.498 sec)
This is a: [[[279.556274 0 231.802872][0 88.2541199 63.5253563][0 0...]]...]
This is a: [[[279.969757 0 231.014099][0 88.2687683 63.6322861][0 0...]]...]
This is a: [[[279.175293 0 232.971161][0 88.2307892 63.7521477][0 0...]]...]
This is a: [[[279.251526 0 232.387619][0 88.2669067 63.9285][0 0...]]...]
This is a: [[[278.991821 0 233.439789][0 88.3050232 64.0183105][0 0...]]...]
This is a: [[[279.318512 0 230.959671][0 88.3264542 64.1823425][0 0...]]...]
This is a: [[[278.350769 0 234.72168][0 88.2409592 64.3132935][0 0...]]...]
This is a: [[[278.289 0 235.590393][0 88.3302917 64.4907455][0 0...]]...]
This is a: [[[277.735901 0 236.21077][0 88.3061371 64.7503][0 0...]]...]
This is a: [[[276.579 0 240.72966][0 88.112793 65.0608368][0 0...]]...]
This is a: [[[277.320343 0 237.961304][0 87.9852 65.5171356][0 0...]]...]
This is a: [[[278.056885 0 235.404724][0 88.084671 65.2708435][0 0...]]...]
This is a: [[[276.114655 0 241.669434][0 87.7980347 66.4421921][0 0...]]...]
This is a: [[[276.216675 0 242.047714][0 87.6519394 66.5866089][0 0...]]...]
This is a: [[[280.178741 0 229.243317][0 88.0968781 65.4064713][0 0...]]...]
This is a: [[[276.885101 0 238.143707][0 87.4040222 67.6298752][0 0...]]...]
This is a: [[[274.371582 0 246.045074][0 86.913681 68.6620102][0 0...]]...]
This is a: [[[271.430023 0 252.620697][0 86.4586716 70.349205][0 0...]]...]
This is a: [[[272.001221 0 249.408981][0 86.4773178 70.3942719][0 0...]]...]
This is a: [[[272.045074 0 247.716187][0 86.5167694 70.1947327][0 0...]]...]
This is a: [[[272.632446 0 246.345871][0 86.5002289 69.6090088][0 0...]]...]
This is a: [[[271.456268 0 248.953979][0 86.2493134 70.0749207][0 0...]]...]
This is a: [[[271.800629 0 247.950363][0 86.1630707 70.2784424][0 0...]]...]
This is a: [[[269.851227 0 252.091217][0 85.9424362 71.2653046][0 0...]]...]
This is a: [[[271.570282 0 247.954727][0 86.0758133 70.2387161][0 0...]]...]
This is a: [[[270.9823 0 249.021164][0 86.0091095 70.0276642][0 0...]]...]
This is a: [[[267.688782 0 257.606262][0 85.6298447 70.4860916][0 0...]]...]
This is a: [[[267.105713 0 258.485474][0 85.5493774 70.4876633][0 0...]]...]
This is a: [[[267.478943 0 255.620285][0 85.6526489 70.9415436][0 0...]]...]
This is a: [[[272.027039 0 244.054688][0 86.0454941 69.6304245][0 0...]]...]
This is a: [[[272.494171 0 241.892075][0 86.0043335 69.73423][0 0...]]...]
This is a: [[[270.740051 0 244.994781][0 85.6205597 70.4630432][0 0...]]...]
This is a: [[[267.617249 0 250.428696][0 85.2291 71.4709][0 0...]]...]
This is a: [[[262.630829 0 262.596527][0 84.4461823 72.8980637][0 0...]]...]
This is a: [[[259.20108 0 271.434326][0 83.7274323 74.0595551][0 0...]]...]
This is a: [[[257.776917 0 273.99649][0 83.2635803 75.6574249][0 0...]]...]
This is a: [[[258.427765 0 270.289581][0 82.9914398 77.1859741][0 0...]]...]
This is a: [[[261.798492 0 257.318237][0 83.3276443 77.7017365][0 0...]]...]
This is a: [[[265.309814 0 244.721909][0 83.8333893 76.3784332][0 0...]]...]
This is a: [[[266.277527 0 239.712845][0 84.0011902 75.6952438][0 0...]]...]
This is a: [[[264.236176 0 244.330215][0 83.8417282 75.6938934][0 0...]]...]
This is a: [[[261.996643 0 254.994629][0 83.7201691 75.3661][0 0...]]...]
This is a: [[[260.339844 0 262.711975][0 83.7451706 75.3466568][0 0...]]...]
This is a: [[[261.646698 0 258.592896][0 84.2305527 74.5424805][0 0...]]...]
This is a: [[[264.99411 0 247.20079][0 84.9026642 72.7443771][0 0...]]...]
This is a: [[[269.234741 0 234.58812][0 85.6514282 70.5073242][0 0...]]...]
This is a: [[[269.672943 0 230.801376][0 85.7210083 70.2081223][0 0...]]...]
This is a: [[[267.528503 0 233.683167][0 85.3473129 71.4683685][0 0...]]...]
